### ISSUE TITLE
Problems with WCAG 2.0 definition of "Web page"

### ISSUE THREAD

**Issue by jasonjgw** [w3c/wcag21#110]
Following a discussion of this topic with Mark Hakkinen, and noting issue #17, the following difficulties have been identified with the current (WCAG 2.0) definition of "Web page".

1. It does not apply to resources made available by means other than HTTP. For example, an HTML/CSS/JavaScript-based document or application stored on a local file system would not qualify as a "Web page" per the definition. Nor would an HTML document that is automatically generated by a user agent, as in the "configuration page" of a browser. Nor would a mobile application composed entirely of HTML, CSS and JavaScript (i.e., implemented using paradigmatic examples of Web technologies). We maintain, however, that such resources should be capable of qualifying as Web pages for purposes of applying WCAG. Removing the "via HTTP" restriction from the definition would resolve this concern.

2. A resource containing a bundle of further resources (e.g., a zip archive, an EPUB publication - see issue #17 - which is also a form of zip archive, etc.), if interpreted by a user agent, would appear to qualify as a "Web page" rather than as a "set of Web pages" under the definition. Note that the resources need not be rendered concurrently, as explained in the notes to the definition, and thus the fact that they would be rendered sequentially does not prevent the definition of "Web page" from being applicable. However, for the purpose of applying WCAG Success Criteria, it is preferable to treat the archive (the bundle) as though it were replaced by its contents - the resource or set of resources which it contains. We need to add a note, or a clause in the definition that addresses this problem.

3. The term "non-embedded" (used in the definition of "Web page") could benefit from clarification. For instance, is an HTML document embedded if it is imported into antoher HTML document and never linked to as a stand-alone resource? Does whether a resource is embedded depend on how it is used/referenced by other resources that lie within the scope of conformance?

We are keen to resolve problems 1 and 2 above, and would welcome clarification of problem 3. In general, I don't think the definitions of "Web page" and "set of Web pages" are the right place in which to impose arbitrary restrictions upon what types of resources WCAG can meaningfully be applied to. Rather, I think those who achieve and then claim conformance should have greater flexibility in deciding in which cases it makes sense to apply WCAG and, if they so choose, to assert conformance in a conformance claim.

**Comment by awkawk** [comment]
As this is comment on WCAG 2.0 and we aren't changing the WCAG 2.0 text in 2.1 we are closing this one and marking "defer" for future consideration. If you have suggested clarification related to item 3 please feel free to suggest text changes for the understanding document.
